src/image/image_type.cpp

```diff
@@ -7,6 +7,19 @@
 
 namespace turi{
 
+image_type::image_type(boost::gil::rgb8_image_t gil_image) {
+  auto it = view(gil_image).begin();
+  auto data = (uint8_t*) &boost::gil::at_c<0>(*it);
```

Comment:
We should get in the habit of using the more proper and less error-prone C++ cast operators.  In this case, you want to use reinterpret_cast<const uint8_t*>(&boost::gil::at_c<0>(*it));